docs/sources/flow/reference/components/remote.git.md

```diff
@@ -0,0 +1,120 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/components/remote.git/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/remote.git/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/remote.git/
+- /docs/grafana-cloud/send-data/agent/flow/reference/components/remote.git/
+canonical: gits://grafana.com/docs/agent/latest/flow/reference/components/remote.git/
+description: Learn about remote.git
+title: remote.git
+---
+
+# remote.git
+
+`remote.git` retrieves the content of file stored in a git repository and exposes it to other components. The repo
+is polled for changes so that the most recent content is always available.
+
+## Usage
+
+```river
+remote.git "LABEL" {
+  repository = "GIT_REPOSITORY"
+  path       = "PATH_TO_FILE"
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`repository` | `string` | The Git repository address to retrieve the content from. | | yes
+`revision` | `string` | The Git revision to retrieve the content from. | `"HEAD"` | no
+`path` | `string` | The path in the repository where the content is stored. | | yes
+`pull_frequency` | `duration` | The frequency to pull the repository for updates. | `"60s"` | no
+
+The `repository` attribute must be set to a repository address that would be
+recognized by Git with a `git clone REPOSITORY_ADDRESS` command, such as
+`htts://github.com/grafana/agent.git`.
+
+The `revision` attribute, when provided, must be set to a valid branch, tag, or
+commit SHA within the repository.
+
+The `path` attribute must be set to a path which is accessible from the root of
+the repository, such as `FILE_NAME.river` or `FOLDER_NAME/FILE_NAME.river`.
+
+If `pull_frequency` is not `"0s"`, the Git repository will be pulled for
+updates at the frequency specified.
```

Comment:
If it is `0s`, will it be pulled once only? or never?

Reply:
it is pulled once

Reply:
sorry, I wasn't asking for an answer, I wanted to suggest adding this to docs